src/ExpressiveInstaller/OptionalPackages.php

```diff
@@ -461,71 +473,68 @@ public static function addPackage(IOInterface $io, $packageName, $packageVersion
-            unset(self::$composerDefinition['require'][$packageName]);
-            unset(self::$composerRequires[$packageName]);
+        if (in_array($packageName, $this->config['require-dev'])) {
+            unset($this->composerDefinition['require'][$packageName]);
+            unset($this->composerRequires[$packageName]);
 
-            self::$composerDefinition['require-dev'][$packageName] = $packageVersion;
-            self::$composerDevRequires[$packageName]               = $link;
+            $this->composerDefinition['require-dev'][$packageName] = $packageVersion;
+            $this->composerDevRequires[$packageName]               = $link;
         } else {
-            unset(self::$composerDefinition['require-dev'][$packageName]);
-            unset(self::$composerDevRequires[$packageName]);
+            unset($this->composerDefinition['require-dev'][$packageName]);
+            unset($this->composerDevRequires[$packageName]);
 
-            self::$composerDefinition['require'][$packageName] = $packageVersion;
-            self::$composerRequires[$packageName]              = $link;
+            $this->composerDefinition['require'][$packageName] = $packageVersion;
+            $this->composerRequires[$packageName]              = $link;
         }
 
         // Set package stability if needed
         switch (VersionParser::parseStability($packageVersion)) {
             case 'dev':
-                self::$stabilityFlags[$packageName] = BasePackage::STABILITY_DEV;
+                $this->stabilityFlags[$packageName] = BasePackage::STABILITY_DEV;
                 break;
             case 'alpha':
-                self::$stabilityFlags[$packageName] = BasePackage::STABILITY_ALPHA;
+                $this->stabilityFlags[$packageName] = BasePackage::STABILITY_ALPHA;
                 break;
             case 'beta':
-                self::$stabilityFlags[$packageName] = BasePackage::STABILITY_BETA;
+                $this->stabilityFlags[$packageName] = BasePackage::STABILITY_BETA;
                 break;
             case 'RC':
-                self::$stabilityFlags[$packageName] = BasePackage::STABILITY_RC;
+                $this->stabilityFlags[$packageName] = BasePackage::STABILITY_RC;
                 break;
         }
     }
 
     /**
      * Copy a file to its final destination in the skeleton.
      *
-     * @param IOInterface $io
-     * @param string      $projectRoot
-     * @param string      $source Source file.
-     * @param string      $target Destination.
-     * @param bool        $force  whether or not to copy over an existing file.
+     * @param string $source Source file.
+     * @param string $target Destination.
+     * @param bool   $force  whether or not to copy over an existing file.
      */
-    public static function copyFile(IOInterface $io, $projectRoot, $source, $target, $force = false)
+    public function copyFile($source, $target, $force = false)
```

Comment:
Just thinking... Can we rename source to resource? Because that's what it does, it copies resources to a target destination.

So the function name would be `copyResource` and `$source`would become `$resource`.

I think it also indicates better that it copies files from `src/ExpressiveInstaller/Resources/`.